pkg/plugin/plugin_init.go

```diff
@@ -95,15 +99,13 @@ func InitPlugin(pluginName string, newEventHandler EventHandlerCreator, newServe
 
 	pluginServer := newServer(webhookSecret, handler)
 	metrics, errs := server.RegisterMetrics(githubClient)
-	if len(errs) > 0 {
-		errLog := logger
-		for _, e := range errs {
-			errLog = errLog.WithError(e)
-		}
-		errLog.Fatal("Prometheus metrics registration failed!")
-	}
+	logErrors(errs, logger, "Prometheus metrics registration failed!")
+
 	pluginServer.Metrics = metrics
 
+	errors := registerMetrics()
+	logErrors(errors, logger, fmt.Sprintf("Prometheus metrics registration failed for plugin %s!", pluginName))
+
```

Comment:
is it possible to register the plugin specific metrics as part of the service creation?
there would be a few benefits
* we would get rid of calling similar method/function twice
* we would need to have the additional method parameter
* the plugins that don't use any metric wouldn't need to create the dummy function: https://github.com/arquillian/ike-prow-plugins/pull/187/files#diff-3bfa5d1e5db805dd35fac5a9d5724c61R51

Reply:
I didn't get your point, what do you mean by service creation?

Reply:
In the implementation of the `ServerCreator` https://github.com/arquillian/ike-prow-plugins/blob/master/pkg/plugin/plugin_init.go#L53 that is called here https://github.com/arquillian/ike-prow-plugins/blob/master/pkg/plugin/plugin_init.go#L96